src/test/groovy/com/user_service/services/UserAuthenticationContractTest.groovy

```diff
@@ -0,0 +1,148 @@
+package com.user_service.services
+
+import com.user_service.controllers.UserAuthenticationController
+import com.user_service.dtos.request.*
+import com.user_service.dtos.response.*
+import org.springframework.http.HttpStatus
+import spock.lang.Specification
+
+import java.time.LocalDate
+
+class UserAuthenticationContractTest extends Specification {
+
+    UserAuthenticationController userAuthenticationController
+    UserAuthenticationService userAuthenticationService
+
+    def setup() {
```

Comment:
Minor thing about test setup: you can use `@Autowired` and `@MockBean` on class fields to simplify code. https://www.baeldung.com/java-spring-mockito-mock-mockbean